### DISCUSSION TITLE
040-Crd-kafka.yaml file is missing the sequence and is failing

### DISCUSSION THREAD

**Discussion by pashafirdous1** [strimzi/strimzi-kafka-operator#4871] [Q&A]
_Please use this to only for bug reports. For questions or when you need help, you can use the [GitHub Discussions](https://github.com/strimzi/strimzi-kafka-operator/discussions), our [#strimzi Slack channel](https://slack.cncf.io/) or out [user mailing list](https://lists.cncf.io/g/cncf-strimzi-users/topics)._

**Describe the bug**
when you apply the CRD file `040-Crd-kafka.yaml` you get the below error
```
The CustomResourceDefinition "kafkas.kafka.strimzi.io" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```

**To Reproduce**
1. Go to ' /install/cluster-operator/'
2. Create Custom Resource 'kafka CRD'
3. Run command '.kubectl apply -f 040-Crd-kafka.yaml.'
```
The CustomResourceDefinition "kafkas.kafka.strimzi.io" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```

**Expected behavior**
the CRD needs to be applied with the necessary API versions.
also when i downloaded and verfied the file, i see that the sequence is broken.

**Environment (please complete the following information):**
 - Strimzi version: [e.g. main, 0.22.1]
 - Installation method: [e.g. YAML files, ]
 - Kubernetes cluster: [e.g. Kubernetes 1.20 GKE]
 - Infrastructure: [e.g. GKE

**YAML files and logs**

Attach or copy paste the custom resources you used to deploy the Kafka cluster and the relevant YAMLs created by the Cluster Operator.
Attach or copy and paste also the relevant logs.
[040-Crd-kafka.yaml.txt](https://github.com/strimzi/strimzi-kafka-operator/files/6405291/040-Crd-kafka.yaml.txt)]

*To easily collect all YAMLs and logs, you can use our [report script](https://github.com/strimzi/strimzi-kafka-operator/blob/main/tools/report.sh) which will automatically collect all files and prepare a ZIP archive which can be easily attached to this issue.
The usage of this script is:
`./report.sh [--namespace <string>] [--cluster <string>]`*

**Additional context**

**Comment by scholzj** [comment]
As described in the docs, you should not use `kubectl apply` but `kubectl create` / `kubectl replace` with Strimzi 0.22 because `kubectl apply` can handle only resources up to some size.

**Reply by ellieayla** [reply]
Is this limit being reached by virtue of kubectl creating the `last-applied-configuration` annotation (with a giant copy of the CRD serialized)?

kubernetes/kubectl#712

**Reply by scholzj** [reply]
Not sure that is the only reason, the CRD is quite big even without it ... but I think it helps :-o.

Just FYI: This problem affects only 0.22 release.

**Reply by ellieayla** [reply]
Did the CRD get smaller in subsequent releases?

**Reply by scholzj** [reply]
Yes. In 0.22, the CRD was migrating from v1alpha1 and v1beta1 versions to v1beta2 because of the removal of v1beta1 CRD API from Kubernetes 1.22. The versions from 0.23+ are smaller because they have only v1beta2.
